DataFormats/Provenance/src/BranchType.cc

```diff
@@ -95,53 +86,75 @@ namespace edm {
     std::string const idToParameterSetBlobsBranch = "IdToParameterSetsBlobs";
   }  // namespace
 
-  std::string const& BranchTypeToString(BranchType const& branchType) {
-    return ((branchType == InEvent) ? event : ((branchType == InRun) ? run : lumi));
-  }
+  std::string const& BranchTypeToString(BranchType const& branchType) { return branchTypeNames[branchType]; }
 
   std::string const& BranchTypeToProductTreeName(BranchType const& branchType) {
-    return ((branchType == InEvent) ? events : ((branchType == InRun) ? runs : lumis));
+    if (branchType >= InProcess) {
```

Comment:
Given this is a programming error, why not just use `assert`?

Reply:
I will make this change. An assert is better here. Thanks.